File: PlaneTicketingService/PlaneTicketingApi/contracts/flightContract.go
package contracts

import (
	"encoding/json"
	"io"
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type FlightContract struct {
	Id                       primitive.ObjectID
	Start                    time.Time
	End                      time.Time
	DepartureLocation        string
	DestinationLocation      string
	PriceOfTicket            float64
	MaxNumberOfTickets       int
	AvailableNumberOfTickets int
}

func (f *FlightContract) FromJSON(r io.Reader) error {
	e := json.NewDecoder(r)
	return e.Decode(f)
}

File: BookingService/BookingWeb/src/app/reservation/contracts/FlightContract.ts
export class FlightContract {
    Id: string;
    Start: Date;
    End: Date;
    DepartureLocation: string;
    DestinationLocation: string;
    PriceOfTicket: number;
    MaxNumberOfTickets: number;
    AvailableNumberOfTickets: number;
  
    constructor(
      Id: string,
      Start: Date,
      End: Date,
      DepartureLocation: string,
      DestinationLocation: string,
      PriceOfTicket: number,
      MaxNumberOfTickets: number,
      AvailableNumberOfTickets: number
    ) {
      this.Id = Id;
      this.Start = Start;
      this.End = End;
      this.DepartureLocation = DepartureLocation;
      this.DestinationLocation = DestinationLocation;
      this.PriceOfTicket = PriceOfTicket;
      this.MaxNumberOfTickets = MaxNumberOfTickets;
      this.AvailableNumberOfTickets = AvailableNumberOfTickets;
    }
  }